python-sdk/src/astro/files/locations/google/gdrive.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import pathlib
+import urllib.parse
+from typing import Any, Iterator, Sequence
+
+from airflow.providers.google.suite.hooks.drive import GoogleDriveHook
+
+from astro.constants import FileLocation
+from astro.files.locations.base import BaseFileLocation
+
+
+def _quote(s: str) -> str:
+    s = s.replace("'", "\\'")  # Escape single quote characters.
+    return f"'{s}'"  # Quote.
+
+
+def _find_item_id(path_parts: Sequence[str], *, conn: Any) -> str:
```

Comment:
Would it make sense to use the Drive hook's `get_file_id` method?
https://github.com/apache/airflow/blob/main/airflow/providers/google/suite/hooks/drive.py#L151

Reply:
The function needs `folder_id`, so we still needs to call it recursively (or in a loop). It is more similar to one single `conn.files().list()` call in this function, but containing more fields and thus less efficient.